### PR TITLE
Update animals table

### PR DESCRIPTION
**Pull request by Khaled-Alaa-1**
### In this project :

- Add a column species of type string to my animal's tablet

- Inside a transaction update the animal's table by setting the species column to unspecified

- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.

- Update the animals table by setting the species column to Pokemon for all animals that don't have species already set.
252612549-0e9439cf-8b15-4ab6-93f9-a9eabda49966

- Delete all records in the animal's table, then roll back the transaction
252612756-20dc023a-c8c2-435e-bdd2-7fdce05f6920

- Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
252613188-89e866b1-a3eb-4d42-bdf6-f181cc923755

### Write queries to answer the following questions:

- How many animals are there?
252613994-0dccdcef-154d-4e9d-96d8-ffdeb95de487

- How many animals have never tried to escape?
252614221-1af3441d-989c-403e-afcc-e4660f2b0fb2

- What is the average weight of animals?
252614439-2125fff1-d5fa-45ca-bec5-45ddccf5230c

- Who escapes the most, neutered or not neutered animals?
252614661-c932784c-cff7-408e-947f-cf50d309aae5

- What is the minimum and maximum weight of each type of animal?
252615011-765e5a79-bce7-4c49-a94f-48a1d2d8512c

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
252615399-3ae923e6-7695-465c-a0c8-f79df1f64fc4

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -5,3 +5,12 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
   (4, 'Devimon', '2017-05-12', 5, true, 11);
+
+  INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES 
+  (5, 'Charmander', '2020-02-08', 0, false, -11),
+	(6, 'Plantmon', '2022-11-15', 2, true, -5.7),
+	(7, 'Squirtle', '1993-04-02', 3, false, -12.13),
+  (8, 'Angemon', '2005-06-12', 1, true, -45),
+  (9, 'Boarmon', '2005-06-07', 7, true, 20.4),
+  (10, 'Blossom', '1998-08-13', 3, true, 17)
```
Comment (Lucas-Masaba) [inline data.sql]: - [ ] Please add Ditto and its details to the animals table. No Digimon left behind. You also did not close this command with a semi-colon.
image

File: queries.sql [hunk 2]
```diff
@@ -39,4 +39,50 @@ WHERE NAME != 'Gabumon';
...
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species = 'unspecified';
```
Comment (Lucas-Masaba) [inline queries.sql]: - [ ] This should be:
`UPDATE animals SET species = 'pokemon' WHERE species IS NULL;`
You removed unspecified from the species column and so it is not present on your table.

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @Khaled-Alaa-1  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- All SQL is in its respective file :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql [hunk 3]
```diff
@@ -5,3 +5,12 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
   (4, 'Devimon', '2017-05-12', 5, true, 11);
+
+  INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES 
+  (5, 'Charmander', '2020-02-08', 0, false, -11),
+	(6, 'Plantmon', '2022-11-15', 2, true, -5.7),
+	(7, 'Squirtle', '1993-04-02', 3, false, -12.13),
+  (8, 'Angemon', '2005-06-12', 1, true, -45),
+  (9, 'Boarmon', '2005-06-07', 7, true, 20.4),
+  (10, 'Blossom', '1998-08-13', 3, true, 17);
```
Comment (usorfaitheloho) [inline data.sql]: You did well by implementing some of the change's suggested by the previous reviewer, kindly make this correction stated by the previous reviewer that you was omitted:

> * [ ]  Please add Ditto and its details to the animals table. No Digimon left behind.
>   image

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
Hi @Khaled-Alaa-1  :thumbsup:  ,

This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

There are still some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- implemented some changes by previous reviewer ✔️
<br/>
<br/>
<br/>

 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @Khaled-Alaa-1,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message.
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**